stl/inc/flat_map

```diff
@@ -219,218 +212,218 @@ struct _Paring_iterator_provider {
             return _Right + _Off;
         }
 
-        operator typename _Paring_iterator_provider<_KeyContainer, _MappedContainer,
-            _Paring_iterator_kind::_Const>::_Iterator() const
-            requires (_Kind == _Paring_iterator_kind::_Mutable)
+        operator _Const_iterator() const
+            requires (!is_same_v<_MappedIter, _MappedConvIter>)
         {
-            return typename _Paring_iterator_provider<_KeyContainer, _MappedContainer,
-                _Paring_iterator_kind::_Const>::_Iterator{_Key_it, _Mapped_it};
+            return _Const_iterator{_Key_it, _Mapped_it};
         }
 
     private:
-        _Key_iterator_t _Key_it;
-        _Mapped_iterator_t _Mapped_it;
+        _KeyIter _Key_it;
+        _MappedIter _Mapped_it;
     };
 
     _STL_INTERNAL_STATIC_ASSERT(swappable<_Iterator>);
 };
 
 _EXPORT_STD

```

Comment:
Pre-existing: `_Flat_map_base` is marked as `_EXPORT_STD`. Either this is a bug, or it should be TRANSITION (I'm not aware of any compiler bug this would be working around; VSO-1538698 "Better handling for non-exported friend function declarations" doesn't affect classes).

Worth checking all `_EXPORT_STD` occurrences.